lib/internal/modules/esm/loader.js

```diff
@@ -158,27 +158,29 @@ class ESMLoader {
   async getModuleJob(specifier, parentURL, importAssertions) {
     let importAssertionsForResolve;
 
-    // We can skip cloning if there are no user-provided loaders because
-    // the Node.js default resolve hook does not use import assertions.
     if (this.#hooks?.hasCustomLoadHooks) {
       importAssertionsForResolve = {
         __proto__: null,
         ...importAssertions,
       };
+    } else {
+      // We can skip cloning if there are no user-provided loaders.
+      importAssertionsForResolve = importAssertions;
     }
 
-    const { format, url } =
-      await this.resolve(specifier, parentURL, importAssertionsForResolve);
+    const resolveResult = await this.resolve(specifier, parentURL, importAssertionsForResolve);
+    const { url, format } = resolveResult;
+    const resolvedImportAssertions = resolveResult.importAssertions ?? importAssertionsForResolve;
```

Comment:
```suggestion
    const resolvedImportAssertions = resolveResult.importAssertions ?? importAssertions;
```

To avoid relying on accidental mutation of the object passed to the loader?